### ISSUE TITLE
🔦 Feature: GUI based custom query builder

### ISSUE THREAD

**Issue by lublak** [nocodb/nocodb#4336]
### Please confirm if feature request does NOT exists already ?

- [X] I confirm there is no existing issue for this

### Describe the usecase for the feature

Sometimes it would be practical to build complicated views.
Not only to filter columns but to pivot etc.
There should be functions to build a view.

### Suggested Solution

As an example there is excel power query.
![image](https://user-images.githubusercontent.com/44057030/200843159-e04e0937-aeec-4fe9-bfea-bb99279ff57f.png)
![image](https://user-images.githubusercontent.com/44057030/200843206-aa4b2ac6-169b-490d-8067-4dfaebefd097.png)

**Comment by o1lab** [comment]
To breakdown the issue there are couple of things here : creating a view from the UI and showing them in UI. The latter part is possible - you can create view using your fav SQL editor and do a [meta sync](https://docs.nocodb.com/setup-and-usages/sync-schema).

**Comment by lublak** [comment]
@o1lab  True. Then it would only be the first part that would have to be made possible. Whereby here also the thought was to make the something simple. Sql works but is rather complicated for most people. Query Builder is mainly like a gui editor. Many functionalities would only be very complicated to realize via sql.
It was just a thought that came to me today when I was working with excel. That such a function would also be very practical in nocodb.
Basically I would not say now that ne own language should be developed. But a sandbox with javascript I would be interesting: https://github.com/patriksimek/vm2
A mixture of knex commands and others could be realized here.

```js
var pivot = tables({ a: 'table', b: 'table' }).select({
  aTitle: 'a.title',
  bTitle: 'b.title'
}).jsonata('jsonata query').pivot(pivot.columns().remove('one columns'));
var cloned = pivot.clone();
pivot.removeColumns(pivot.columns());
return cloned;
```

**Comment by o1lab** [comment]
Ok, I see your point. However, generally we would like to avoid code like knex within nocodb as it will be too narrow and limiting for people who are not familiar with js or knex.

**Comment by lublak** [comment]
@o1lab The basic idea is this. Alternatively, there would also be flow-based editing, for example.
Example from n8n:
![image](https://user-images.githubusercontent.com/44057030/200857428-f3bb862d-f0a5-4533-8b51-529e0db9586d.png)

**Comment by mmospanenko** [comment]
Will be awesome to have something like Metabase inside for analytics, reporting
